fastparquet/util.py

```diff
@@ -193,9 +220,7 @@ def metadata_from_many(file_list, verify_schema=False, open_with=default_open,
                     rg = copy.copy(rg)
                     rg.columns = [copy.copy(c) for c in rg.columns]
                     for chunk in rg.columns:
-                        chunk.file_path = '/'.join(
-                            [fn, chunk.file_path if isinstance(chunk.file_path, str) else chunk.file_path.decode()]
-                        )
+                        chunk.file_path = '/'.join([fn, chunk.file_path])
```

Comment:
I changed at `ParquetFile` instantiation the decoding to string behavior.
In current version of fastparquet, only the `file_path` of the 1st column is decoded to string.
This is seen in `_parse_header()` (current version below).
```python
        for rg in fmd[4]:
            chunks = rg[1]
            if chunks:
                chunk = chunks[0]
                s = chunk.get(1)
                if s:
                    chunk[1] = s.decode()
```
Now, included in this PR, all columns are dealt with:
```python
        # for rg in fmd.row_groups:
        for rg in fmd[4]:
            # chunks = rg.columns
            chunks = rg[1]    
            if chunks:
                for chunk in chunks:
                    # s = chunk.file_path
                    s = chunk.get(1)
                    if s:
                        # chunk.file_path = s.decode()
                        chunk[1] = s.decode()
```
To be noticed, there was a related `TODO` in current code, in `writer.find_max_part()`.
```python
def find_max_part(row_groups):
    """
    Find the highest integer matching "**part.*.parquet" in referenced paths.
    """
    paths = [c.file_path or "" for rg in row_groups for c in rg.columns]
    # TODO: the following line should not be necessary
    paths = [p.decode() if isinstance(p, bytes) else p for p in paths]
```
I confirm that after the change in logic, previously related failing test cases are now ok.

Reply:
Ah, this might be an unfortunate change then. Decoding the bytes can take a non-trivial amount of time when there are many columns, and in the typical read-only case, it's unnecessary.